src/structs.rs

```diff
@@ -74,12 +98,32 @@ pub struct Delay {
     pub delay: f64,
 }
 
+pub struct PoisonDelay {
+    pub start: Instant,
+    pub ticks: i8
```

Comment:
why i8 ?

Reply:
Because it's just used as a counter. I guess usize would be better.